src/NHapi.Base/Model/AbstractSegment.cs

```diff
@@ -26,468 +26,510 @@
+        {
+            Add(c, required, maxReps, length, constructorArgs, null);
+        }
+
+        [Obsolete("This method has been replaced by 'Add'.")]
+        [System.Diagnostics.CodeAnalysis.SuppressMessage(
+            "StyleCop.CSharp.NamingRules",
+            "SA1300:Element should begin with upper-case letter",
+            Justification = "As this is a public member, we will duplicate the method and mark this one as obsolete.")]
+        protected internal virtual void add(
+            Type c,
+            bool required,
+            int maxReps,
+            int length,
+            object[] constructorArgs,
+            string description)
+        {
+            Add(c, required, maxReps, length, constructorArgs, description);
+        }
+
+        /// <summary>
+        /// Add a segment.
+        /// </summary>
+        /// <param name="c">The type of segment.</param>
+        /// <param name="required"></param>
+        /// <param name="maxReps"></param>
+        /// <param name="length"></param>
+        /// <param name="constructorArgs"></param>
+        /// <param name="description"></param>
+        protected internal virtual void Add(
+            Type c,
+            bool required,
+            int maxReps,
+            int length,
+            object[] constructorArgs,
+            string description)
+        {
+            if (!typeof(IType).IsAssignableFrom(c))
+            {
+                throw new HL7Exception(
+                    "Class " + c.FullName + " does not inherit from " + "ca.on.uhn.datatype.Type",
+                    ErrorCode.APPLICATION_INTERNAL_ERROR);
+            }
+
+            items.Add(new AbstractSegmentItem(c, required, maxReps, length, constructorArgs, description));
+        }
+
+        /// <summary> Creates a new instance of the Type at the given field number in this segment.  </summary>
+        private IType CreateNewType(int field)
+        {
+            int number = field - 1;
+            Type c = items[number].FieldType;
+
+            var description = items[number].Description;
+
+            IType newType = null;
+            try
+            {
+                object[] args = GetArgs(number, description);
+                Type[] argClasses = new Type[args.Length];
+                for (int i = 0; i < args.Length; i++)
+                {
+                    if (args[i] is IMessage)
+                    {
+                        argClasses[i] = typeof(IMessage);
+                    }
+                    else
+                    {
+                        argClasses[i] = args[i].GetType();
+                    }
+                }
+
+                newType = (IType)c.GetConstructor(argClasses).Invoke(args);
+            }
+            catch (UnauthorizedAccessException iae)
+            {
+                throw new HL7Exception(
+                    "Can't access class " + c.FullName + " (" + iae.GetType().FullName + "): " + iae.Message,

```

Comment:
if there is a quick way to prefer string interpilation ... lets do it :D

Reply:
done